src/db/memory.rs

```diff
@@ -9,55 +9,65 @@ use anyhow::Result;
 use cid::Cid;
 use fvm_ipld_blockstore::Blockstore;
 use parking_lot::RwLock;
+use std::ops::Deref;
 
-use super::{Error, Store};
+use super::SettingsStore;
 
-/// A thread-safe `HashMap` wrapper.
+type MemoryDbInner = Arc<RwLock<HashMap<Vec<u8>, Vec<u8>>>>;
+
+/// A thread-safe `HashMap` wrapper, acting as a memory-backed blockstore.
 #[derive(Debug, Default, Clone)]
-pub struct MemoryDB {
-    db: Arc<RwLock<HashMap<Vec<u8>, Vec<u8>>>>,
+pub struct MemoryDB(MemoryDbInner);
+
+impl Deref for MemoryDB {
```

Comment:
I guess, `Deref` is usually for smart pointer types. We can instead add an `into_inner()` which is standard or just do `.0`

Reply:
It's a leftover. Removed.